python/dgl/dataloading/dataloader.py

```diff
@@ -514,10 +517,12 @@ def __init__(self, sample_func, g, use_uva, device):
         self.device = device
 
     def __call__(self, items):
-        if self.use_uva or (self.g.device != torch.device('cpu')):
-            # Only copy the indices to the given device if in UVA mode or the graph is not on
-            # CPU.
-            items = recursive_apply(items, lambda x: x.to(self.device))
+        if self.use_uva:
+            graph_device = getattr(self.g, 'device', None)
```

Comment:
If this because graph storage may not have a `device` attribute?

Reply:
Yes.